### PR TITLE
MINOR: remove zk from several tests

### PR DESCRIPTION
**Pull request by cmccabe**
Remove ZK from
 - core/src/test/scala/integration/kafka/admin/RemoteTopicCrudTest.scala
 - core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala
 - core/src/test/scala/unit/kafka/server/DeleteTopicsRequestTest.scala
 - tools/src/test/java/org/apache/kafka/tools/consumer/group/SaslClientsWithInvalidCredentialsTest.java

### CODE REVIEW

File: core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala [hunk 1]
```diff
@@ -191,8 +182,9 @@ class AddPartitionsTest extends BaseRequestTest {
     assertEquals(Set(0, 1), replicas.asScala.toSet)
   }
 
+  @Disabled
   @ParameterizedTest
-  @ValueSource(strings = Array("zk")) // TODO: add kraft support
+  @ValueSource(strings = Array("kraft")) // TODO: add kraft support
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala]: We need to revisit this test to make it stable under KRaft. I've opened Jira https://issues.apache.org/jira/browse/KAFKA-18098 to address it. Could you please revert this change?

File: core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala [hunk 2]
```diff
@@ -217,8 +209,9 @@ class AddPartitionsTest extends BaseRequestTest {
     validateLeaderAndReplicas(topicMetadata, 6, 0, Set(0, 1, 2, 3))
   }
 
+  @Disabled
   @ParameterizedTest
-  @ValueSource(strings = Array("zk")) // TODO: add kraft support
+  @ValueSource(strings = Array("kraft")) // TODO: add kraft support
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala]: ditto

**Review by chia7712 — COMMENTED** [review]
@cmccabe thanks for this patch. Could you please fix the conflicts?

File: core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala (hunk 1 above)
Comment (peterxcli) [reply core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala]: Hi @chia7712, thanks for addressing this.
PR is ready! https://github.com/apache/kafka/pull/17955

Comment (cmccabe) [reply core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala]: sure, let's address this in #17955. I'll remove it from this PR. Thanks @peterxcli and @chia7712

File: core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala (hunk 2 above)
Comment (cmccabe) [reply core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala]: sure, let's address this in #17955. I'll remove it from this PR. Thanks @peterxcli and @chia7712

**Comment by cmccabe** [comment]
- Removed modifications to `core/src/test/scala/unit/kafka/admin/AddPartitionsTest.scala` and removed it from the PR description
- Added reference to `core/src/test/scala/integration/kafka/server/GssapiAuthenticationTest.scala` to the PR description.

**Review by chia7712 — APPROVED** [review]

File: tools/src/test/java/org/apache/kafka/tools/consumer/group/SaslClientsWithInvalidCredentialsTest.java [hunk 3]
```diff
@@ -106,13 +115,13 @@ public void setUp(TestInfo testInfo) {
         startSasl(jaasSections(KAFKA_SERVER_SASL_MECHANISMS, Some$.MODULE$.apply(KAFKA_CLIENT_SASL_MECHANISM), Both$.MODULE$,
             JaasTestUtils.KAFKA_SERVER_CONTEXT_NAME));
         super.setUp(testInfo);
```
Comment (chia7712) [inline tools/src/test/java/org/apache/kafka/tools/consumer/group/SaslClientsWithInvalidCredentialsTest.java:117]: This fails with latest code ... I have filed https://issues.apache.org/jira/browse/KAFKA-18109 to fix it.
